common/include/pcl/point_cloud.h

```diff
@@ -426,17 +395,17 @@ namespace pcl
       std::vector<PointT, Eigen::aligned_allocator<PointT> > points;
 
       /** \brief The point cloud width (if organized as an image-structure). */
-      std::uint32_t width;
+      std::uint32_t width = 0;
       /** \brief The point cloud height (if organized as an image-structure). */
-      std::uint32_t height;
+      std::uint32_t height = 0;
 
       /** \brief True if no points are invalid (e.g., have NaN or Inf values in any of their floating point fields). */
-      bool is_dense;
+      bool is_dense = true;
```

Comment:
Note: `PCLPointCloud2` is initialized with `is_dense` set to false but this one is set to `true` :/

Reply:
Not nice, but probably also not critical. I think we should change the default in `PCLPointCloud2` because hardly anyone constructs that type of point cloud "by hand".

Reply:
Can you confirm what you wrote? You state one thing but your argument seems to point in the other direction. I interpreted what you wrote as

> I think we should **not** change the default in `PCLPointCloud2` because hardly anyone constructs that type of point cloud "by hand".

`PCLPointCloud2` comes from ROS messages and these usually means actual sensors which naturally have nans, while the `PointClouds` are very often handcrafted. Which would actually mean, this was a well thought decision and their difference is intentional.

Reply:
No, I indeed meant what I wrote. My thought is that since `PCLPointCloud2` comes from ROS message, the _default-constructed_ value of this field does not make any difference, it will be set to whatever value the message had. Similarly, when you call `toPCLPointCloud2()` to create a blob point cloud,`is_dense` will be set to that of the source cloud.